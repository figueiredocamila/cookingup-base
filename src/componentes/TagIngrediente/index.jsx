import { styled } from "styled-components"

const Botao = styled.button`
    background: ${props => props.$ativo ? 'var(--coral, #F0633C)' : 'var(--cinza-claro, #EFEFEF)'};
    color: ${props => props.$ativo ? 'var(--creme, #FFFAF3)' : 'var(--cinza, #444)'};
    font-weight: ${props => props.$ativo ? '700' : '400'};
    padding: 8px;
    border-radius: 8px;
    border: none;
    cursor: pointer;
    text-align: center;
    font-family: Nunito Sans;
    font-size: 18px;
    font-style: normal;
    line-height: 150%; 
`


const TagIngrediente = ({ ingrediente, onClick, ativo }) => {
    return (<Botao $ativo={ativo} onClick={onClick}>
        {ingrediente.nome}
    </Botao>)
}

export default TagIngrediente