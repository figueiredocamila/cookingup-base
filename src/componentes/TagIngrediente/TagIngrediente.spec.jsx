import { fireEvent, render, screen } from "@testing-library/react";
import { expect, test, vi } from "vitest";
import TagIngrediente from "./index";

const ingredienteMock = {
  nome: "Tomate",
};


test("TagIngrediente component renders correctly", () => {
  render(<TagIngrediente ingrediente={ingredienteMock} ativo={false} />);
  const buttonElement = screen.getByRole("button", { name: /Tomate/i });
  expect(buttonElement).toBeInTheDocument();
})

test("should call onClick when the button is clicked", () => {
  const onClickMock = vi.fn();

  const { getByRole } = render(<TagIngrediente ingrediente={ingredienteMock} ativo={false} onClick={onClickMock} />);

  const buttonElement = getByRole("button", { name: /Tomate/i });

  fireEvent.click(buttonElement);
  expect(onClickMock).toHaveBeenCalledTimes(1);
});

test("should have the 'ativo' class when the 'ativo' prop is true", () => {
  const { getByRole } = render(<TagIngrediente ingrediente={ingredienteMock} ativo={true} />);

  const buttonElement = getByRole("button", { name: /Tomate/i });

  expect(buttonElement).toMatchSnapshot();
});

test("should not have the 'ativo' class when the 'ativo' prop is false", () => {
  const { getByRole } = render(<TagIngrediente ingrediente={ingredienteMock} ativo={false} />);

  const buttonElement = getByRole("button", { name: /Tomate/i });

  expect(buttonElement).toMatchSnapshot();
});