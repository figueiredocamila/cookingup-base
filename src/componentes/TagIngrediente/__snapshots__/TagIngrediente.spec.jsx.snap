// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should have the 'ativo' class when the 'ativo' prop is true 1`] = `
<button
  class="sc-bczSft kUJMIt"
>
  Tomate
</button>
`;

exports[`should not have the 'ativo' class when the 'ativo' prop is false 1`] = `
<button
  class="sc-bczSft eYMIVd"
>
  Tomate
</button>
`;
